client/modules/IDE/components/SketchListRowBase.jsx

```diff
@@ -0,0 +1,167 @@
+import PropTypes from 'prop-types';
+import slugify from 'slugify';
+import React, { useState, useRef, useCallback } from 'react';
+import { Link } from 'react-router-dom';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import * as ProjectActions from '../actions/project';
+import * as IdeActions from '../actions/ide'; // Adding the missing Ide actions
+import TableDropdown from '../../../components/Dropdown/TableDropdown';
+import MenuItem from '../../../components/Dropdown/MenuItem';
+import dates from '../../../utils/formatDate';
+import getConfig from '../../../utils/getConfig';
+
+const ROOT_URL = getConfig('API_URL');
+
+const formatDateCell = (date, mobile = false) =>
+  dates.format(date, { showTime: !mobile });
+
+const SketchListRowBase = ({
+  sketch,
+  username,
+  user,
+  changeProjectName,
+  cloneProject,
+  deleteProject,
+  t,
+  mobile,
+  onAddToCollection
+}) => {
+  const [renameOpen, setRenameOpen] = useState(false);
+  const [renameValue, setRenameValue] = useState(sketch.name);
+  const renameInput = useRef(null);
+
+  const openRename = useCallback(() => {
+    setRenameOpen(true);
+    setRenameValue(sketch.name);
+    renameInput.current.focus();
+  }, [sketch.name]);
+
+  const closeRename = () => setRenameOpen(false);
+
+  const updateName = useCallback(() => {
+    if (renameValue.trim().length > 0) {
+      changeProjectName(sketch.id, renameValue.trim());
+    }
+  }, [renameValue, sketch.id, changeProjectName]);
+
+  const handleRenameChange = (e) => setRenameValue(e.target.value);
+
+  const handleRenameEnter = (e) => {
+    if (e.key === 'Enter') {
+      e.preventDefault();
+      updateName();
+      closeRename();
+    }
+  };
+
+  const handleRenameBlur = () => {
+    updateName();
+    closeRename();
+  };
+
+  const handleSketchDownload = () => {
+    const downloadLink = document.createElement('a');
+    downloadLink.href = `${ROOT_URL}/projects/${sketch.id}/zip`;
+    downloadLink.download = `${sketch.name}.zip`;
+    document.body.appendChild(downloadLink);
+    downloadLink.click();
+    document.body.removeChild(downloadLink);
+  };
+
+  const handleSketchDuplicate = () => cloneProject(sketch);
+  const handleSketchDelete = () => {
+    if (window.confirm(t('Common.DeleteConfirmation', { name: sketch.name }))) {
+      deleteProject(sketch.id);
+    }
+  };
+
+  const userIsOwner = user.username === username;
+
+  let url = `/${username}/sketches/${sketch.id}`;
+  if (username === 'p5') {
+    url = `/${username}/sketches/${slugify(sketch.name, '_')}`;
+  }
+
+  const name = (
+    <>
+      <Link to={url}>{renameOpen ? '' : sketch.name}</Link>
+      {renameOpen && (
+        <input
+          value={renameValue}
+          onChange={handleRenameChange}
+          onKeyDown={handleRenameEnter}
+          onBlur={handleRenameBlur}
```

Comment:
was comparing this to the original component and noticed there was an event.stopPropagation on that one -- do we need this here?

Reply:
Good point! Added to the component :)